src/hooks/useFrictionGroup.ts

```diff
@@ -35,7 +50,7 @@ export const useFrictionGroup = <E extends HTMLElement | SVGElement = any>(
       // Derive interpolator functions for the supplied CSS properties.
       const interpolators = getInterpolatorsForPairs(
         {
-          from: props.from,
+          from: cache.get(hookId.current)?.[i] ?? props.from,
```

Comment:
❓ What if `n` changes? The cache lookup key could be wrong based on index?

💭 add `n` to the key somehow? and/or a `useEffect` that cleans/resets cache if `n` changes?

Reply:
Good question and good idea! Let me give you a full picture of the caching setup now.

So in this PR we create a global cache of animation values, which allows us to track these values for an arbitrary number of elements created by an arbitrary number of `useFriction` / `useFrictionGroup` hooks in your React tree. It basically looks like this:

```js
new Map([
  [
    // The first instantiated hook in the tree.
    'frictionGroup_0', {
      // Current animation values for the first element in the first hook in tree.
      0: {
        opacity: 0.50,
        transform: 'translateX(20px)'
      },
      // Current animation values for the second element in the first hook in tree.
      1: {
        opacity: 0.40,
        transform: 'translateX(15px)'
      }
  ],
  [
    'frictionGroup_1',
    {
      // Current animation values for the first element in the second hook in tree.
      0: {
        transform: 'scale(0.75)'
      }
    }
  ],
]);
```

Looking at it now, I think this cache is actually more information than we need. Each hook can be responsible for creating its own local cache of animation values that would get cleaned up when the component using it unmounts.

This then would allow us to directly use your idea of combining `n` and `i` into a cache key quite nicely! My only thought is that this would mean that adding / removing elements by altering `n` would force all animations to start from their `from` config (since they'd have no cached entry), which may or may not be the experience we want compared to adding / removing elements only affecting the added / removed element and allowing the others to continue animating fluidly. I'm going to run some experiments on `react-spring` and see how it works over there.

Great shout @hartzis, appreciate you taking the time to review!

Reply:
Following up here, `react-spring` appears to fully restart all animations whenever n changes: https://codesandbox.io/s/zealous-borg-2ykl8?file=/src/App.tsx, so that's at least a good starting place for us to think about what we want our behavior to be. It's likely a bit safer to do that, but I want to have a longer think (outside the scope of this PR though).